geomstats/geometry/cholesky.py

```diff
@@ -0,0 +1,324 @@
+"""The manifold of lower triangular matrices with positive diagonal elements"""
+
+import math
+
+import geomstats.backend as gs
+from geomstats.geometry.base import OpenSet
+from geomstats.geometry.general_linear import GeneralLinear
+from geomstats.geometry.matrices import Matrices
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+from geomstats.geometry.lower_triangular_matrices import LowerTriangularMatrices
+from geomstats.geometry.lie_group import MatrixLieGroup
+
+class CholeskySpace(OpenSet):
+    """Class for the manifold of lower triangular matrices with positive diagonal elements.
+
+    Parameters
+    ----------
+    n : int
+        Integer representing the shape of the matrices: n x n.
+    """
+
+    def __init__(self, n, **kwargs):
+        super(CholeskySpace, self).__init__(
+            dim=int(n * (n + 1) / 2),
+            metric=(CholeskyMetric),
+            ambient_space=LowerTriangularMatrices(n), **kwargs)
+        self.n = n
+
+    def belongs(self, mat, atol=gs.atol):
+        """Check if a matrix is lower triangular matrix with 
+        positive diagonal elements
+
+        Parameters
+        ----------
+        mat : array-like, shape=[..., n, n]
+            Matrix to be checked.
+        atol : float
+            Tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean denoting if mat is an SPD matrix.
+        """
+        is_lower_triangular = self.ambient_space.belongs(mat, atol)
+        diagonal = gs.diag(mat)
+        is_positive = gs.all(diagonal > 0, axis=-1)
+        belongs = gs.logical_and(is_lower_triangular, is_positive)
+        return belongs
+
+    def projection(self, point):
+        """Project a matrix to the cholesksy space.
+
+        First it is projected to space lower triangular matrices
+        and then diagonal elements are exponentiated to make it positive
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Matrix to project.
+
+        Returns
+        -------
+        projected: array-like, shape=[..., n, n]
+            SPD matrix.
+        """
+        vec_diag = gs.exp(gs.diagonal(point, axis1 = -2, axis2 = -1))
+        diag = gs.vec_to_diag(vec_diag)
+        strictly_lower_triangular = Matrices.to_lower_triangular(point)
+        projection = diag + strictly_lower_triangular
+        return projection
```

Comment:
Ideally, the projection should minimize the Frobenius norm. This is tricky as we have an open set here, but one property that must be verified for a projection function p is p(p(x))=x for all x.
This is not the case here, so maybe only do the projection if point does not already belong to LT+, or use epsilon to floor negative values of the diagonal.

Reply:
@nguigs nice catch! I forgot we should we have p(p(x))=x. I will just use take `abs` of the diagonal